### DISCUSSION TITLE
What is '/run/user/1000/libpod/tmp/exits' and how to clean it up?

### DISCUSSION THREAD

**Discussion by djarbz** [containers/podman#17931] [General]
`/run/user/1000` is a space limited tmpfs and I recently found that it filled up due to the files in `/run/user/1000/libpod/tmp/exits` that seem to only contain an exit code. I have almost 200k of these files.

Is there a best practice for cleaning these files up?

**Comment by Luap99** [comment]
How many containers do you run on this host (or restart the container very often)?
Podman should remove this file when the container is stopped, see https://github.com/containers/podman/blob/78f1ebb80ba473f4f9bb016aec5d190645089cce/libpod/container_internal.go#L752-L759
If the file is leaking it sounds like you are hitting some error before the file is removed.

Also what version are you using, I suggest you try with the latest podman if you are not using it already.

**Comment by djarbz** [comment]
I have about 30 containers on this host.

```
Client:       Podman Engine
Version:      4.3.1
API Version:  4.3.1
Go Version:   go1.19.4
Built:        Wed Dec 31 18:00:00 1969
OS/Arch:      linux/amd64
```

```
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
```

**Reply by gee456** [reply]
I have a similar problem with /run/libpod/persist/ filling up with folders that have a file called exit in them. The code for exit is 0 so why aren't they being cleaned?

**Reply by gee456** [reply]
To provide more context about my situation. We use CEPH with the cephadm container orchestration tool. It creates and removes containers daily. We discovered it because we started to run out of inodes on /run/libpod/persist.
My Temp fix is to create a daily cron job that finds completed logs more than 15 days old and purges them
Here is the script if someone wants to use it 

```
fullcount=$(find /run/libpod/persist -name "exit" -print | wc -l) 
filecount=$(find /run/libpod/persist -name "exit" -mtime +15 -print | wc -l)
logger "LIBPODCLEANER: Cleaning $filecount of $fullcount /run/libpod/persist folders that are over 15 days old"
for i in $(ls /run/libpod/persist);do  
if ls /run/libpod/persist/$i | grep -q exit; then    
  if grep -q 0 /run/libpod/persist/$i/exit; then   
    if [[ $(find "/run/libpod/persist/$i/exit" -mtime +15 -print &>/dev/null) ]]; then  
       #echo deleting $i  
       rm -rf /run/libpod/persist/$i   
    fi  
  fi  
fi  
done  
fullcount=$(find /run/libpod/persist -name "exit" -print | wc -l)  
logger "LIBPODCLEANER: complete $fullcount folders still present /run/libpod/persist"  

```
[taistejlaiho](https://github.com/taistejlaiho) Opened a bug request here to fix the issue https://github.com/containers/podman/issues/25291
